src/main/java/duke/command/CommandProcessor.java

```diff
@@ -0,0 +1,188 @@
+package duke.command;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import duke.codec.TimeProcessor;
+import duke.exceptions.InputException;
+import duke.exceptions.ProcessingException;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * The `CommandProcessor` class handles the processing of user commands in the Duke application.
+ * It interprets user input, executes commands, and interacts with the storage system to manage tasks.
+ */
+public class CommandProcessor {
+    private final Storage storage;
+
+    public CommandProcessor(Storage storage) {
+        this.storage = storage;
+    }
+
+    /**
+     * Processes the given command and input, executing the corresponding action.
+     *
+     * @param command The command to execute.
+     * @param input   The user input associated with the command.
+     * @throws InputException      If there is an issue with the input format or command.
+     * @throws ProcessingException If there is an issue processing the command or storage.
+     */
+    public String processData(Command command, String input) throws InputException, ProcessingException {
+        String result = "";
+        switch (command) {
+
+        case DELETE:
+            result = storage.delete(processDelete(input));
+            break;
```

Comment:
Good to see using the right indentation for the switch/case block.

Reply:
Thanks!